src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,138 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (head == null) {
+            head = tail = newNode;
+        } else {
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexToAdd(index);
+        Node<T> thatNode = new Node<>(value);
+        if (head == null) {
+            head = tail = thatNode;
+        } else if (index == 0) {
+            thatNode.next = head;
+            head = thatNode;
+        } else if (index == size) {
+            tail.next = thatNode;
+            tail = thatNode;
+        } else {
+            Node<T> prev = getNodeByIndex(index - 1);
+            thatNode.next = prev.next;
+            prev.next = thatNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T lis : list) {
```

Comment:
variable name lis is a bit confusion, you could replace it with something like listElement